src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,136 @@
+import classNames from 'classnames';
+import React, {
+  ChangeEvent, FormEvent, useEffect, useRef,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoLoader } from '../TodoLoader';
+
+type Props = {
+  todo: Todo;
+  isProcessing: boolean;
+  onRemoveTodo?: (todoId: number) => void;
+  onUpdateStatus?: (todo: Todo) => void;
+  onSetEditTodoId?: (todoId: number | null) => void;
+  editTodoId?: number | null;
+  onSetNewTitle?: (title: string) => void;
+  newTitle?: string;
+  onUpdateTitle?: (todo: Todo, cancel?: boolean) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onRemoveTodo,
+  isProcessing,
+  onUpdateStatus,
+  onSetEditTodoId,
+  editTodoId,
+  onSetNewTitle,
+  newTitle,
+  onUpdateTitle,
+}) => {
+  const { title, completed, id } = todo;
+  const editTitle = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editTitle.current) {
+      editTitle.current.focus();
+    }
+  }, [editTodoId]);
+
+  const handleUpdateStatusInput = () => {
+    if (onUpdateStatus) {
+      onUpdateStatus(todo);
+    }
+  };
```

Comment:
use `useEffect` for all functions.

Reply:
To be honest i don`t understand what do you mean.